### PR TITLE
Add seqan::min function for simd vectors

### PR DESCRIPTION
**Pull request by marehr**
Add seqan::min function for simd vectors.

### CODE REVIEW

File: tests/simd/test_simd_vector.h [hunk 1]
```diff
@@ -441,6 +441,27 @@ SEQAN_TYPED_TEST(SimdVectorTestCommon, Max)
...
+
+    TSimdVector a{0u}, b{0u};
+    fillVectors(a, b);
+
+    auto c = min(a, b);
+
+    for (auto i = 0; i < length; ++i)
+    {
+        TValue a_i = -3 + i * 3, b_i = length - i;
+        // std::cout << i << " / " << length << ": " << (int)c[i] << " = min (" << (int)a[i] << ", " << (int)b[i] << ")" << std::endl;
```
Comment (h-2) [inline tests/simd/test_simd_vector.h]: can this line be removed?

Comment (marehr) [reply tests/simd/test_simd_vector.h]: I have this line in all test functions.

I can remove them in all those functions, but it was quite handy to have them to comment them out, if something was wrong. It is only a test case.

Comment (h-2) [reply tests/simd/test_simd_vector.h]: Hm, but it is not clear for someone who doesn't know the code. Commented out code can mean anything: "this broke and I couldn't fix it so I just removed it", "this is debug and has no effect", "I forgot to re-add this and it is necessary" ....
So I would remove it in 99% of the test cases. If you have very complex code that is super-hard to debug, and you wrote some debugging functionality, you could leave it inside, but then I would instead:
```cpp
#if 0 // debug(marehr): The following code block does ....
    foobar();
    bax();
    i + 3;
#endif
```
So it is absolutely clear that it was intentionally kept, by whom, and why.

Comment (marehr) [reply tests/simd/test_simd_vector.h]: @h-2, a `std::cout`, which does not change any state and is in a test case, is clearly for debugging.
I read your concern and I agree with you in general, but in this case it adds zero complexity, but is really handy to have when something fails.

If this is your only remark, could please merge it?

Comment (rrahn) [reply tests/simd/test_simd_vector.h]: I suggest meet half way and simply add a DEBUG info at the beginning of the std::cout output: 
like so: `// std::cout << "DEBUG:" << ... << std::endl;`

**Pull request merged**